File: frontends/frontend-qooxdoo/dc2-qooxdoo/source/class/dc2/dialogs/search/Hosts.js
/*
    (DC)² - DataCenter Deployment Control
    Copyright (C) 2010, 2011, 2012  Stephan Adig <sh@sourcecode.de>

    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation; either version 2 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License along
    with this program; if not, write to the Free Software Foundation, Inc.,
    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
*/

qx.Class.define("dc2.dialogs.search.Hosts",
{
  extend: dc2.widgets.DialogWidget,
  construct:function() {
    this.base(arguments);
  },
  members: {
    _edit_hostname:null,
    _edit_domainname:null,
    _createLayout:function() {
      var comp=new qx.ui.container.Composite();
      var layout=new qx.ui.layout.Grid(5,10);
      layout.setColumnFlex(1,1);
      comp.setLayout(layout);
      this._edit_hostname=new qx.ui.form.TextField();
      this._edit_domainname=new qx.ui.form.TextField();
      comp.add(new qx.ui.basic.Label("Hostname"),{row:0,column:0});
      comp.add(new qx.ui.basic.Label("Domainname"),{row:1,column:0});
      comp.add(this._edit_hostname,{row:0,column:1});
      comp.add(this._edit_domainname,{row:1,column:1});
      return(comp);
    },
    _getData:function() {
      var data={};
      if (this._edit_hostname.getValue() != null && this._edit_hostname.getValue() != "") {
        data["hostname"]=this._edit_hostname.getValue();
      }
      if (this._edit_domainname.getValue() != null && this._edit_domainname.getValue() != "") {
        data["domainname"]=this._edit_domainname.getValue();
      }
      return(data);
    }
  }
});

File: components/dc2-api/dc2/api/dc2/inventory/hosts.py
# -*- coding: utf-8 -*-
###############################################################################
#
#    (DC)² - DataCenter Deployment Control
#    Copyright (C) 2010, 2011, 2012, 2013, 2014  Stephan Adig <sh@sourcecode.de>
#    This program is free software; you can redistribute it and/or modify
#    it under the terms of the GNU General Public License as published by
#    the Free Software Foundation; either version 2 of the License, or
#    (at your option) any later version.
#
#    This program is distributed in the hope that it will be useful,
#    but WITHOUT ANY WARRANTY; without even the implied warranty of
#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
#    GNU General Public License for more details.
#
#    You should have received a copy of the GNU General Public License along
#    with this program; if not, write to the Free Software Foundation, Inc.,
#    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
###############################################################################

import types
from dc2.api import RPCClient


class Hosts(RPCClient):

    def find(self, rec=None):
        hostlist = []
        if rec is None:
            hostlist = self._proxy.dc2.inventory.hosts.list()
            return hostlist
        if rec is not None:
            if type(rec) is not types.DictType:
                # TODO: Add Real Exception
                raise Exception('The search argument is not a dictionary')
            hostlist = self._proxy.dc2.inventory.hosts.find(rec)
        return hostlist

    def list(self):
        hostlist = self.find()
        return hostlist

    def count(self):
        # TODO: Add a rpc call to appserver for counting
        hostlist = self.find()
        return len(hostlist)

    def get(self, *args, **kwargs):
        host = {}
        if 'id' in kwargs:
            host['_id'] = kwargs.get('id', None)
        if 'server_id' in kwargs:
            host['server_id'] = kwargs.get('server_id', None)
        if len(host) != 0:
            host_entries = self._proxy.dc2.inventory.hosts.find(host)
            if host_entries is not None and len(host_entries) == 1:
                return host_entries[0]
        return None

    def update(self, *args, **kwargs):
        host = {}
        if 'host' in kwargs:
            host = kwargs.get('host', None)
        doc_id = self._proxy.dc2.inventory.hosts.update(host)
        return doc_id
